src/_micro-api/libraries/signalDecoder/src/signalDecoder.cpp

```diff
@@ -315,10 +207,7 @@ void SignalDetectorClass::compress_pattern()
 
 
 				int  sum = histo[idx] + histo[idx2];
-				if (sum == 0)
-					pattern[idx] = (long(pattern[idx]) * histo[idx] / sum) + (pattern[idx2] * histo[idx2] / sum);
-				else
-					pattern[idx] = (long(pattern[idx]) + pattern[idx2]) / 2;
+				pattern[idx] = ((long(pattern[idx]) * histo[idx]) + (pattern[idx2] * histo[idx2])) / sum;
```

Comment:
im alten Code war eine. In den Schleifen ist immer ein continue, wenn histo 0 ist.